packages/snowpack/src/commands/build.ts

```diff
@@ -52,25 +51,192 @@ async function installOptimizedDependencies(
   return installResult;
 }
 
+/**
+ * FileBuilder - This class is responsible for building a file. It is broken into
+ * individual stages so that the entire application build process can be tackled
+ * in stages (build -> resolve -> write to disk).
+ */
+class FileBuilder {
+  output: Record<string, string> = {};
+  filesToResolve: Record<string, SnowpackSourceFile> = {};
+  filesToProxy: string[] = [];
+
+  constructor(
```

Comment:
nit: should these be an object instead? I’d be worried about accidentally mixing up a `string` type def on a new instantiation

Reply:
I'm game :+1: